Classes/common/CDTDatastore.m

```diff
@@ -150,6 +168,11 @@ - (NSUInteger)documentCount
 
 - (NSString *)name { return self.database.name; }
 
+- (id<CDTEncryptionKeyProvider>)copyEncryptionKeyProvider
+{
+    return [_encryptionKeyProvider copy];
+}
```

Comment:
When a datastore is initialised, [I copy the key provider passed as a parameter](https://github.com/cloudant/CDTDatastore/pull/112/files#diff-09611c62c3ef826c53341a9070bc3d7dR70). I do that to ensure that the key does not change once the datastore is created. For this same reason, I have to return a copy and not the key provider stored as a property.

Given that I return a copy (retain count +1), I have to prefix the method with `copy` to follow the [naming convention](http://clang.llvm.org/docs/AutomaticReferenceCounting.html#method-families).